processor/src/main/java/org/mapstruct/ap/internal/util/MapperConfiguration.java

```diff
@@ -170,12 +170,15 @@ private boolean isMapToDefaultOnMapperAndMappingConfigLevel() {
 
 
     public String componentModel() {
-        if ( mapperConfigPrism != null && mapperPrism.values.componentModel() == null ) {
-            return mapperConfigPrism.componentModel();
-        }
-        else {
+        if ( mapperPrism.values.componentModel() != null ) {
             return mapperPrism.componentModel();
         }
+
+        if ( mapperConfigPrism != null && mapperConfigPrism.values.componentModel() != null ) {
```

Comment:
Okay, `MapperConfig#unmappedTargetPolicy` is ignored if there was a value specified as processor option. Updating the PR...